packages/vscode-extension/extension.js

```diff
@@ -1,77 +1,98 @@
-// The module 'vscode' contains the VS Code extensibility API
-// Import the module and reference it with the alias vscode in your code below
 const vscode = require('vscode');
-
-// this method is called when your extension is activated
-// your extension is activated the very first time the command is executed
-
-// const storeTime = (projectname, filename, starttime) => {
-// 	const endTime = Date.now();
-
-// }
+const { saveActivities } = require('./storage/server.js');
 
 class ActivefileWatcher {
 	/**
 	 * @param {number} interval Polling interval
-	 *
 	 */
-	constructor(interval = 1000) {
+	constructor(interval = 1000, changeCallback) {
 		this.startTime = null;
 		this.activefile = null;
 		this.activeProject = null;
 		this.language = null;
-		// this.changeCallback = changeCallback;
+		this.gitBranch = null;
+		this.remoteUrl = null;
+		this.projectPath = null;
+		this.changeCallback = changeCallback;
 		this.interval = interval;
+		this.intervalId = null;
 	}
 
 	storeTime() {
 		const endTime = Date.now();
 		const startTime = this.startTime;
-		const projectname = this.activeProject;
-		const filename = this.activefile;
+		const projectName = this.activeProject;
+		const fileName = this.activefile;
 		const languageId = this.language;
+		const gitBranch = this.gitBranch;
+		const remoteUrl = this.remoteUrl;
+		const projectPath = this.projectPath;
 		const data = {
-			projectname,
-			filename,
+			projectName,
+			projectPath,
+			fileName,
 			languageId,
+			gitBranch,
+			remoteUrl,
 			startTime,
 			endTime,
 		};
 
-		console.log(data);
+		this.changeCallback(data);
 	}
 	tracker() {
-		setInterval(() => {
+		this.intervalId = setInterval(() => {
+			let currentProjectPath = vscode.workspace.workspaceFolders[0].uri.path;
```

Comment:
Found another edge case. 
![image](https://user-images.githubusercontent.com/44284655/150749186-0b24c128-70f1-4a92-8172-c9d08aa4bdfc.png)

`vscode.workspace.workspaceFolders` can be undefined.
![image](https://user-images.githubusercontent.com/44284655/150749543-2869bb46-f422-441b-94ee-ea9fe49a1a82.png)